code_folder/22_data_extraction_pipeline.py

```
@@ -0,0 +1,232 @@
+
+
+# ---------------------------------------------------------------------------
+# 2. Extract sklearn Built-in Dataset — California Housing
+# ---------------------------------------------------------------------------
+def extract_california_housing() -> pd.DataFrame:
+    """Load California Housing dataset from sklearn."""
+    print("[Extract] Loading California Housing from sklearn")
+    housing = fetch_california_housing(as_frame=True)
+    df = housing.frame
+    print(f"  -> {df.shape[0]} rows, {df.shape[1]} columns")
+    print(f"  -> Target: MedHouseVal (median house value in $100k)")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)